test/evergreen/code_coverage_analysis.py

```diff
@@ -22,23 +23,84 @@ def read_timing_data(timing_data_path: str):
         print("Timing data: {} to {}, delta {}".format(start_time_secs, end_time_secs, delta_secs))
         return delta_secs
 
+# Calculate the per component coverage from the detailed files.
+def get_component_coverage_list(branch_coverage, coverage_files, outfile):
+    component_dict = defaultdict(list)
+    new_component_dict = {}
+
+    if outfile is None:
+        outfile='atlas_out_code_coverage.json'
+
+    # Using for loop
+    for i in coverage_files:
+        component = i["filename"].split('/')[1]
+        if component not in component_dict:
+            component_dict[component] = [i["branch_covered"], i["branch_total"]]
+        else:
+            component_dict[component][0] += i["branch_covered"]
+            component_dict[component][1] += i["branch_total"]
+
+    # Insert the overall branch coverage data
+    new_component_dict["overall"] = branch_coverage
+
+    for key in component_dict:
+        percent =  100 * float(component_dict[key][0]) / float(component_dict[key][1])
+        new_component_dict[key] = percent
+
+    resultList = []
+    for key, value in new_component_dict.items():
+        atlas_format = {}
+        atlas_format["name"] = key
+        atlas_format["value"]=value
+        resultList.append(atlas_format)
+
+    return (resultList)
+
+# Generate the Atlas compatible format report.
+def get_component_coverage(branch_coverage, coverage_files, outfile):
+    atlas_format = {
+                'Test Name': "Code Coverage",
+                'config': {},
+                'metrics': get_component_coverage_list(branch_coverage, coverage_files, outfile),
+            }
+
+    dir_name = os.path.dirname(outfile)
+    if dir_name:
+        os.makedirs(dir_name, exist_ok=True)
+
+    with open(outfile, 'w') as outfile:
+        json.dump(atlas_format, outfile, indent=4, sort_keys=True)
+
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument('-c', '--coverage', required=True, help='Path to the gcovr json summary data file')
+    parser.add_argument('-s', '--summary', required=True, help='Path to the gcovr json summary data file')
+    parser.add_argument('-o', '--outfile', help='path of the file to write test output to')
```

Comment:
path --> Path (for consistency).

Reply:
Fixed in 8f643b8384ce228db1a05bad5b326e67a3713d8c